lib/new_relic/agent/new_relic_service.rb

```diff
@@ -33,7 +33,7 @@ class NewRelicService
       CONNECTION_ERRORS = [Timeout::Error, EOFError, SystemCallError, SocketError].freeze
 
       attr_accessor :request_timeout
-      attr_reader :collector, :marshaller, :agent_id
+      attr_reader :collector, :marshaller, :agent_id, :request_headers_map
```

Comment:
hmmm....why here?

Reply:
I added this here so that we could access it the same way we are accessing the `agent_id` for the metadata as well. 
`NewRelic::Agent.agent.service.request_headers_map`
`NewRelic::Agent.agent.service.agent_id.to_s`.

Reply:
Alright as we discussed last week, i removed that from the main agent, and switched to using `instance_variable_get` in the infinite tracing connection.